### ISSUE TITLE
[React Wrappers] attributeMapping htmlFor

### ISSUE THREAD

**Issue by pascalvos** [break-stuff/cem-tools#53]
customElementReactWrapperPlugin
```
  attributeMapping: {
        'for': 'htmlFor',
  },
```
this would work if not that all keys are lowercase so htmlFor becomes htmlfor

if I manually do this 

```
export const MyLabel = forwardRef(
  ({ level, htmlFor, id, className, style, slot, hidden, key, children, onClick }, forwardedRef) => {
    const ref = useRef(null);

    /** Attributes - run whenever an attr has changed */
    useAttribute(ref, "level", level);
    useAttribute(ref, "for", htmlFor);
    useAttribute(ref, "id", id);
    useAttribute(ref, "style", style);
    useAttribute(ref, "slot", slot);
    useAttribute(ref, "hidden", hidden);
    useAttribute(ref, "key", key);
    useAttribute(ref, "children", children);
    useAttribute(ref, "ref", ref);

    useImperativeHandle(forwardedRef, () => ({}));

    return React.createElement(
      MyLabelElement.customTag || "my-label",
      {
        ref,
        level,
        for: htmlFor,
        id,
        class: className,
        style,
        slot,
        hidden,
        key,
        children,
        ref,
        onClick,
      },
      children,
    );
  },
);
```

```
export interface MyLabelProps {
  /** Level the label, determines the text size. Highel level has smaller text. */
  level?: MyLabelElement["level"];

  /** Id of the input to which this label should be linked. */
  htmlFor?: MyLabelElement["htmlFor"];
  ```

now everything works as expected when I do this

```
<MyLabel htmlFor="textInput">textInput</UcLabel>
<TextInput name="textInput" id="textInput" />
```

**Comment by pascalvos** [comment]
nevermind i found out https://github.com/break-stuff/cem-tools/blob/main/tools/utilities/src/utilities.ts#L39 this works 
```
    attributeMapping: {
        'for': 'html-for',
      },
```

I do think for and htmlFor might be a good candidate for in here
https://github.com/break-stuff/cem-tools/blob/main/packages/react-wrappers/src/global.ts

**Comment by break-stuff** [comment]
That's a really good idea! Thank you!

**Comment by break-stuff** [comment]
A fix for this has been deployed in release 1.2.0 (https://github.com/break-stuff/cem-tools/pull/67). Let me know if you run into any issues.

**Comment by pascalvos** [comment]
@break-stuff found a small oversight

```
var createEventName = (event) => `on${toPascalCase(event.name)}`;
var RESERVED_WORDS = [
... here the "for", attribute is still added so it will throw an error
```

**Comment by break-stuff** [comment]
Does your `for` attribute differ from the native HTML `for` attribute behavior?

**Comment by pascalvos** [comment]
well for in react doesnt work that only works with htmlFor https://legacy.reactjs.org/docs/dom-elements.html#htmlfor so there is this thing right react is all jsx so javascript so a `for` loop is not same as `for` HTML attribute. 
i see no way that that for loop could end up in the cem tools so I think... you can assume its always an attribute
but not 100% sure I think for should be automatically mapped to htmlFor :)

**Comment by break-stuff** [comment]
You are correct. Because these components render in JSX, `for` attributes should always be mapped to something other than `for`. you can use the default `htmlFor` or you can create a custom mapping: `for` => `_for`.

**Comment by break-stuff** [comment]
I deployed a fix for this in 1.4.0. Let me know if you run into any issues.

**Comment by break-stuff** [comment]
Closing this for now. Let me know if anything comes up.
